web/js/modules/product-picker/reducers.js

```diff
@@ -1,42 +1,186 @@
+import googleTagManager from 'googleTagManager';
+
 import {
-  UPDATE_PRODUCT_PICKER,
-  UPDATE_LIST_SCROLL_TOP,
+  SAVE_SEARCH_STATE,
+  INIT_SEARCH_STATE,
+  COLLAPSE_FACET,
+  SELECT_CATEGORY,
+  SELECT_MEASUREMENT,
+  SELECT_SOURCE,
+  SELECT_LAYER,
+  SHOW_MEASUREMENTS,
+  TOGGLE_FEATURED_TAB,
+  TOGGLE_SEARCH_MODE,
+  TOGGLE_CATEGORY_MODE,
+  TOGGLE_MOBILE_FACETS,
   RESET_STATE,
 } from './constants';
 
-const projToListType = {
-  arctic: 'measurements',
-  antarctic: 'measurements',
-  geographic: 'category',
-};
-
 export const productPickerState = {
-  listType: 'category',
+  mode: 'category',
   category: undefined,
-  filteredRows: [],
-  searchResultRows: undefined,
-  numRowsFilteredOut: undefined,
-  inputValue: '',
-  filterByAvailable: true,
-  listScrollTop: 0,
+  categoryType: 'hazards and disasters',
+  filters: [],
+  showMobileFacets: true,
+  searchTerm: '',
   selectedLayer: undefined,
   selectedMeasurement: undefined,
   selectedMeasurementSourceIndex: 0,
+  searchConfig: undefined,
+  collapsedFacets: {},
 };
 
 export function getInitialState(config) {
-  return { ...productPickerState, categoryType: Object.keys(config.categories)[1] };
+  return productPickerState;
 }
 
 export function productPickerReducer(state = productPickerState, action) {
   switch (action.type) {
-    case UPDATE_PRODUCT_PICKER:
-      return { ...state, ...action.value };
-    case UPDATE_LIST_SCROLL_TOP:
-      return { ...state, listScrollTop: action.value };
+    case INIT_SEARCH_STATE: {
+      const { searchConfig } = action;
+      return {
+        ...state,
+        searchConfig,
+      };
+    }
+
+    case SAVE_SEARCH_STATE: {
+      const { filters, searchTerm } = action;
+      return {
+        ...state,
+        filters,
+        searchTerm,
+        searchConfig: null,
+      };
+    }
+
+    case COLLAPSE_FACET: {
+      const { field } = action;
+      const { collapsedFacets } = state;
+      return {
+        ...state,
+        collapsedFacets: {
+          ...state.collapsedFacets,
+          [field]: !collapsedFacets[field],
+        },
+      };
+    }
+
+    case SELECT_CATEGORY: {
+      googleTagManager.pushEvent({
```

Comment:
I like how you brought this outside of the react views. **dev meeting topic**  gtm middleware. https://github.com/rangle/redux-beacon (I keep forgetting to bring this up)

Reply:
Ooo this looks interesting